examples/cookies_example.rs

```diff
@@ -2,30 +2,55 @@
 extern crate nickel_cookies;
 extern crate cookie;
 
-use nickel_cookies::Cookies;
-use nickel::{Nickel, HttpRouter, QueryString};
-use cookie::Cookie;
+use nickel_cookies::{Cookies, KeyProvider};
+use nickel::{Nickel, HttpRouter, QueryString, Request, Response, MiddlewareResult};
+use cookie::{Cookie, Key};
+
+fn secure_middleware<'mw, 'conn, D: KeyProvider>(req: &mut Request<'mw, 'conn, D>, mut res: Response<'mw, D>) -> MiddlewareResult<'mw, D> {
+    let old_value = { // block for borrow management
+        let server_key = res.server_data().key();
+        let key = Key::from_master(&server_key.0);
```

Comment:
Would it make sense to change the return type from `.key()` to what cookie requires here?

Reply:
Change it from `SecretKey` to `[u8; 32]`? That might work. Would that mean `SecretKey` no longer needs to be public? That would reduce the chance of name-space collision, which seems like a small win too.

Reply:
I was thinking that you could change the return type to whatever `cookies::Key` is? Is that already `[u8; 32]`?

Reply:
`cookie::Key` does not exist if the secure feature is not enabled. We might able to work around that, but that looks to be some fairly extensive changes I'd prefer to bundle into a different PR.